server/src/main/java/org/opensearch/telemetry/TelemetrySettings.java

```diff
@@ -66,10 +81,44 @@ public void setSamplingProbability(double samplingProbability) {
         this.samplingProbability = samplingProbability;
     }
 
+    /**
+     * Set sampling ratio for action
+     * @param filters map of action and corresponding value
+     */
+    public void setActionSamplingProbability(Map<String, Double> filters) {
+        synchronized (this) {
```

Comment:
Can we do the atomic switch?

Reply:
We can't use atomic switch here as filters doesn't contains all value. It only contains one that are passed in update call